diff --git mlibc-clean/abis/linux/statvfs.h mlibc-workdir/abis/linux/statvfs.h
index 7a75d3a..15269b0 100644
--- mlibc-clean/abis/linux/statvfs.h
+++ mlibc-workdir/abis/linux/statvfs.h
@@ -23,6 +23,8 @@ struct statvfs {
 	unsigned long f_fsid;
 	unsigned long f_flag;
 	unsigned long f_namemax;
+
+	char f_basetype[80];
 };
 
 #endif /* _ABIBITS_STATVFS_H */
diff --git mlibc-clean/options/ansi/generic/time-stubs.cpp mlibc-workdir/options/ansi/generic/time-stubs.cpp
index 887a7d3..7fd46bc 100644
--- mlibc-clean/options/ansi/generic/time-stubs.cpp
+++ mlibc-workdir/options/ansi/generic/time-stubs.cpp
@@ -242,6 +242,7 @@ size_t strftime(char *__restrict dest, size_t max_size,
 			c++;
 			break;
 		}
+		case 'l':
 		case 'I': {
 			int hour = tm->tm_hour;
 			if(!hour)
diff --git mlibc-clean/options/glibc/generic/execinfo.cpp mlibc-workdir/options/glibc/generic/execinfo.cpp
index 3474615..e034881 100644
--- mlibc-clean/options/glibc/generic/execinfo.cpp
+++ mlibc-workdir/options/glibc/generic/execinfo.cpp
@@ -1,9 +1,10 @@
 #include <execinfo.h>
 #include <bits/ensure.h>
+#include <mlibc/debug.hpp>
 
 int backtrace(void **, int) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+	mlibc::infoLogger() << "libc: backtrace() is not implemented!\n" << frg::endlog;
+	return 0;
 }
 
 char **backtrace_symbols(void *const *, int) {
